src/app/modules/OpenshiftStreams/dialogs/CreateInstance/CreateInstanceWithSizes/hooks/useAMSQuota.ts

```diff
@@ -0,0 +1,108 @@
+import { InstanceType } from '@app/utils';
+
+/**
+ * Hook that fetches available entitelemts from AMS
+ *
+ * @returns
+ */
+export const useAMSQuota = () => {
+  const config = useConfig();
+  const auth = useAuth();
+
+  // TODO we need this details to be shared to from app-servies-ui
+  const quotaProductId = 'RHOSAK';
+  const trialQuotaProductId = 'RHOSAKTrial';
+  const resourceName = 'rhosak';
+
+  return async () => {
+    const accessToken = await auth?.ams.getToken();
+    const ams = new AppServicesApi({
+      accessToken,
+      basePath: config?.ams.apiBasePath || '',
+    } as Configuration);
+
+    const account = await ams.apiAccountsMgmtV1CurrentAccountGet();
+    const orgId = account?.data?.organization?.id;
+    const quotaData = new Map<QuotaType, QuotaValue>();
+    // TODO remove service down and other values. Use different model?
+    const filteredQuota: Quota = { loading: true, isServiceDown: false, data: undefined };
+
+    if (!orgId) {
+      console.error('useQuota', 'orgId is not defined');
+      filteredQuota.loading = false;
+      filteredQuota.isServiceDown = true;
+      return filteredQuota;
+    }
+
+    try {
+      const response = await ams.apiAccountsMgmtV1OrganizationsOrgIdQuotaCostGet(orgId, undefined, true);
+
+      const quota = response?.data?.items?.find((q) =>
+        q.related_resources?.find((r) => r.resource_name === resourceName && r.product === quotaProductId)
+      );
+      const trialQuota = response?.data?.items?.find((q) =>
+        q.related_resources?.find((r) => r.resource_name === resourceName && r.product === trialQuotaProductId)
+      );
+
+      // TODO logic here should include marketplace vs standard billing model.
+      // We need to pick standard
+      if (quota && quota.allowed > 0) {
+        const remaining = quota?.allowed - quota?.consumed;
+        quotaData?.set(QuotaType.kas, {
+          allowed: quota?.allowed,
+          consumed: quota?.consumed,
+          remaining: remaining < 0 ? 0 : remaining,
+        });
+      }
+
+      if (trialQuota) {
+        quotaData?.set(QuotaType.kasTrial, {
+          allowed: trialQuota?.allowed,
+          consumed: trialQuota?.consumed,
+          remaining: trialQuota?.allowed - trialQuota?.consumed,
+        });
+      }
+
+      filteredQuota.loading = false;
+      filteredQuota.data = quotaData;
+    } catch (error) {
+      filteredQuota.loading = false;
+      filteredQuota.isServiceDown = true;
+      console.error(error);
+    }
+
+    return filteredQuota;
+  };
+};
+
+// Helper methods for busines logic handled in the UI
+export function convertQuotaToInstanceType(quota: Map<QuotaType, QuotaValue> | undefined) {
+  let kasQuota: QuotaValue | undefined;
+  try {
+    kasQuota = quota?.get(QuotaType?.kas);
+  } catch (e) {
+    console.error('useAvailableProvidersAndDefault', 'quota?.get exception', e);
+  }
+  if (kasQuota !== undefined && kasQuota.remaining > 0) {
```

Comment:
```suggestion
  if (kasQuota !== undefined && kasQuota.remaining >= 0) {
```
I think InstanceType should be standard if user has quota `kasQuota.remaining`. @wtrocki correct me if I'm wrong.
`kasQuota.remaining` count could be 0 if user consumed all the limit then still InstanceType  should be standard. 

InstanceType `developer` should be only when user don't have standard quota .

Reply:
I actually copied that code from other place. Not sure if remaining should be used. 
Curent BU requirement is to not allow trial instances to be created even if user runs out of quota. 

kasQuota.remaining >= 0 will be always true as there is no possibility of negative values. I think we might need to remove it.

Reply:
This logic reflects very well bu requirement: https://github.com/bf2fc6cc711aee1a0c2a/kas-ui/blob/4842ab5d11df30a91af77d6e275e0adabc7473db/src/app/modules/OpenshiftStreams/dialogs/CreateInstance/api.ts#L207-L209